addons/timecop/bootstrap.js

```diff
@@ -0,0 +1,40 @@
+const {classes: Cc, interfaces: Ci, utils: Cu} = Components;
+Cu.import("resource://gre/modules/Services.jsm");
+
+const BROWSER_SEARCH_PREF = "browser.search.";
+
+let gInstall = false;
+
+function install(aData, aReason) {
+  // We check for install here instead of in startup because
+  // In Firefox 46, the install reason didn't work properly
+  // in startup in some cases.
+  if (aReason != 5 /* ADDON_INSTALL */) {
+    return;
+  }
+  gInstall = true;
+}
+
+function uninstall(aData, aReason) {}
+
+function startup(aData, aReason) {
+  if (!gInstall) {
+    return;
+  }
+  Services.prefs.setBoolPref(BROWSER_SEARCH_PREF + "reset.geo", true);
+}
+
+function shutdown(aData, aReason) {
+  if (aReason != 2 /* APP_SHUTDOWN */) {
+    return;
+  }
+  try {
+    let geoReset = Services.prefs.getBoolPref(BROWSER_SEARCH_PREF + "reset.geo", false);
+    if (geoReset) {
+      Services.prefs.clearUserPref(BROWSER_SEARCH_PREF + "region");
```

Comment:
Is there any reason a user would have intentionally set this? I'm not sure how these are exposed.

Reply:
No, these are set automatically behind the scenes. A user would never have set this themselves.